src/definitions/bluebps.json

```diff
@@ -0,0 +1,276 @@
+                ]
+            },
+            "current_timeout": {
+                "id": "9103",
+                "description": "Get the over-current timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If the battery current exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "temperature_max": {
+                "id": "9104",
+                "description": "Get the maximum allowed battery temperature",
+                "payload": [
+                    {
+                        "name": "limit",
+                        "type": "u16",
+                        "description": "The minimum voltage allowed for any individual cell. 0~5000: 0~5V",
+                        "units": "millivolt"
+                    }
+                ]
+            },
+            "temperature_timeout": {
+                "id": "9105",
+                "description": "Get the over-temperature timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If the battery temperature exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "state": {
+                "id": "9106",
+                "description": "Get the current state of the device",
+                "payload": [
+                    {
+                        "name": "battery_voltage",
+                        "type": "u16",
+                        "description": "The main battery voltage",
+                        "units": "millivolt"
+                    },
+                    {
+                        "name": "battery_current",
+                        "type": "u16",
+                        "description": "The current measurement",
+                        "units": "centiampere"
+                    },
+                    {
+                        "name": "battery_temperature",
+                        "type": "u16",
+                        "description": "The battery temperature",
+                        "units": "decidegrees Centigrade"
+                    },
+                    {
+                        "name": "cpu_temperature",
+                        "type": "u16",
+                        "description": "The cpu temperature",
+                        "units": "decidegrees Centigrade"
+                    },
+                    {
+                        "name": "flags",
+                        "type": "u8",
+                        "description": "flags indicating if any of the configured limits are currently exceeded"
```

Comment:
Is there a description of each bit flag ?

Reply:
Not yet, but there will be. The flags have not been decided yet in firmware.

Reply:
Flags will be things like 'over current', 'leak', 'high temperature' etc.